tests/tests-memcached/src/main/java/com/hazelcast/simulator/memcached/ReadWriteTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.simulator.memcached;
+
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.Prepare;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Teardown;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+
+import java.util.Random;
+
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateByteArray;
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateStrings;
+
+public class ReadWriteTest extends MemcachedTest {
+
+    // properties
+    public int keyCount = 1000;
+    public int keyLength = 10;
+    public int valueCount = 1000;
+    public int minSize = 16;
+    public int maxSize = 2000;
+    // the number of keys that are going to be written.
+    // normally you want to keep this the same as keyCount (for reading), but it can help to expose certain problems like
+    // gc. If they writeKeyCount is very small, only a small group of objects get updated frequently and helps to prevent
+    // getting them tenured. If writeKeyCount is -1, it will automatically be set to keyCount
+    public int writeKeyCount = -1;
+
+    private String[] keys;
+    private byte[][] values;
+
+    @Setup
+    public void setUp() {
+        keys = generateStrings(keyCount, keyLength);
+
+        if (minSize > maxSize) {
+            throw new IllegalStateException("minSize can't be larger than maxSize");
+        }
+
+        if (writeKeyCount == -1) {
+            writeKeyCount = keyCount;
+        }
+    }
+
+    @Prepare
+    public void prepare() {
+        Random random = new Random();
+        values = new byte[valueCount][];
+        for (int i = 0; i < values.length; i++) {
+            int delta = maxSize - minSize;
+            int length = delta == 0 ? minSize : minSize + random.nextInt(delta);
+            values[i] = generateByteArray(random, length);
+        }
+
+        for (String key : keys) {
+            cluster.add(key, Integer.MAX_VALUE, values[random.nextInt(values.length)]);
+        }
+    }
+
+    @TimeStep(prob = 0.1)
+    public void put(ThreadState state) {
```

Comment:
Does cluster.add return the old value, or void?

Reply:
The cluster.add return OperationFuture<Boolean>, something like a Future, because the call is made asynchronous. The return value determines whether the operation succeeded or not. WDYT, should we return the cluster.add() itself or should we return the actual value cluster.add().get()?